### DISCUSSION TITLE
Questions about mh_* tool options

### DISCUSSION THREAD

**Discussion by EmilyGraceSeville7cf** [florianschanda/miss_hit#243] [Q&A]
Hello! I have several questions about `mh_metrics` options:

1. What encodings are exactly supported by `--input-encoding`? How can I list them from terminal?
2. Is it intended `--text`|`--html`|`--json` are not mutually exclusive options for `mh_metrics` (according to output of `--help` option)?
2. If `--portable-html` is planned to be used only along with `--html` why not to describe it as `[--html FILE [--portable-html]]` instead of `[--html FILE] [--portable-html]`?

**Comment by florianschanda** [comment]
>What encodings are exactly supported by --input-encoding? How can I list them from terminal?

Anything that https://docs.python.org/3/library/codecs.html supports; but realistically I would only ever consider two options: utf-8 and ascii (and maybe the windows cp1252)

> Is it intended --text|--html|--json are not mutually exclusive options for mh_metrics (according to output of --help option)?

They should be.

> If --portable-html is planned to be used only along with --html why not to describe it as [--html FILE [--portable-html]] instead of [--html FILE] [--portable-html]?

I am limited by what argparse can do; but I will look into it if it can group stuff together like this. I haven't read the docs in a while :)

That said, I am looking at changing the way HTML generation works, so that --portable-html will probably disappear entirely.

**Comment by EmilyGraceSeville7cf** [comment]
It is not clear for me what is expected for `--change-entity` (`mh_copyright` tool):

> --change-entity OLD_COPYRIGHT_HOLDER

Can you clarify it? I mean what OLD_COPYRIGHT_HOLDER type is (any string or some file to read from). I don't know what I have to [suggest](https://github.com/fish-shell/fish-shell/pull/8499) to user.

The same misunderstanding I feel with `--primary-entity COPYRIGHT_HOLDER`.

**Reply by florianschanda** [reply]
So the manual provides a small example:


Replaces the old entity with the primary copyright holder. Leaves year ranges unchanged. For example:

```
% (c) Copyright 1998-2020 Old Company GmbH
```

Will be changed to this:

```
% (c) Copyright 1998-2020 New Company GmbH
```

The relevant invocation here would be:
```
mh_copyright --change-entity "Old Company GmbH" --primary-entity "New Company GmbH"
```

Primary entity is basically the key copyright holder; i.e. you or the current name of your company.
If you change your company name you'd update the primary entity in the config file; and then to fix the existing code you can use --change-entity to just replace all the old names with the current primary entity.

**Comment by florianschanda** [comment]
> I don't know what I have to suggest to user.

@EmilySeville7cfg  I think there is nothing to suggest here. By definition these tasks will be "one time" tasks, and not something obvious or repetitive is going on.
